pkg/clients/release/plans.go

```diff
@@ -43,26 +44,51 @@ func (r *ReleaseController) CreateReleasePlan(name, namespace, application, targ
 }
 
 // CreateReleasePlanAdmission creates a new ReleasePlanAdmission using the given parameters.
-func (r *ReleaseController) CreateReleasePlanAdmission(name, namespace, environment, origin, policy, serviceAccountName string, applications []string, autoRelease bool, pipelineRef *tektonutils.PipelineRef, data *runtime.RawExtension) (*releaseApi.ReleasePlanAdmission, error) {
-	releasePlanAdmission := &releaseApi.ReleasePlanAdmission{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
-			Labels: map[string]string{
-				releaseMetadata.AutoReleaseLabel: strconv.FormatBool(autoRelease),
+func (r *ReleaseController) CreateReleasePlanAdmission(name, namespace, environment, origin, policy, serviceAccountName string, applications []string, autoRelease bool, pipelineRef *tektonutils.PipelineRef, data *runtime.RawExtension, timeouts *tektonv1.TimeoutFields) (*releaseApi.ReleasePlanAdmission, error) {
+	var releasePlanAdmission *releaseApi.ReleasePlanAdmission
+	if timeouts != nil {
+		releasePlanAdmission = &releaseApi.ReleasePlanAdmission{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+				Labels: map[string]string{
+					releaseMetadata.AutoReleaseLabel: strconv.FormatBool(autoRelease),
+				},
 			},
-		},
-		Spec: releaseApi.ReleasePlanAdmissionSpec{
-			Applications: applications,
-			Data:         data,
-			Environment:  environment,
-			Origin:       origin,
-			Pipeline: &tektonutils.Pipeline{
-				PipelineRef:    *pipelineRef,
-				ServiceAccountName: serviceAccountName,
+			Spec: releaseApi.ReleasePlanAdmissionSpec{
+				Applications: applications,
+				Data:         data,
+				Environment:  environment,
+				Origin:       origin,
+				Pipeline: &tektonutils.Pipeline{
+					PipelineRef:    *pipelineRef,
+					ServiceAccountName: serviceAccountName,
+					Timeouts: *timeouts,
+				},
+				Policy: policy,
 			},
-			Policy: policy,
-		},
+		}
+	} else {
+		releasePlanAdmission = &releaseApi.ReleasePlanAdmission{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+				Labels: map[string]string{
+					releaseMetadata.AutoReleaseLabel: strconv.FormatBool(autoRelease),
+				},
+			},
+			Spec: releaseApi.ReleasePlanAdmissionSpec{
+				Applications: applications,
+				Data:         data,
+				Environment:  environment,
+				Origin:       origin,
+				Pipeline: &tektonutils.Pipeline{
+					PipelineRef:    *pipelineRef,
+					ServiceAccountName: serviceAccountName,
+				},
+				Policy: policy,
+			},
+		}
```

Comment:
I think you dont need to repeat the whole RPA just to add the the timeouts or not.

Reply:
Thanks. Updated!